cmake/introspection.cmake

```diff
@@ -159,6 +159,40 @@ check_cxx_source_compiles("
   " HAVE_SYSCTL_ARND
 )
 
+# Check for clock_gettime() (POSIX.1b).
```

Comment:
I don't think we need to introduce checks here, we already use `clock_gettime`. You could just #ifdef `CLOCK_THREAD_CPUTIME_ID` at the callsite.